client/rpc_sender.go

```diff
@@ -109,6 +111,9 @@ func (s *rpcSender) Send(_ context.Context, call proto.Call) {
 		break
 	}
 	if err != nil {
-		call.Reply.Header().SetGoError(err)
+		return nil, proto.NewError(err)
 	}
+	pErr := br.Error
```

Comment:
This pattern bothers me a bit (error on the BatchResponse only used to pass error up from Call), is this one of the things that will be changing?